akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedBehaviorImpl.scala

```diff
@@ -193,6 +185,9 @@ private[akka] final case class EventSourcedBehaviorImpl[Command, Event, State](
     copy(recovery = Recovery(selection.toUntyped))
   }
 
+  override def snapshotWhen(predicate: (State, Event, Long) => Boolean): EventSourcedBehavior[Command, Event, State] =
+    copy(snapshotWhen = predicate)
```

Comment:
Is this meant to be where it would actually be evolvable now?  Where the Long in the predicate could be something other than count?

Reply:
Here the `Long` is always the sequenceNr, and this is not for evolving retention criteria. This is the application domain specific decision that it's a good time to save a snapshot, e.g. when BookingCompleted